lib/src/main/java/ECSE428Project/model/Match.java

```diff
@@ -0,0 +1,110 @@
+package ECSE428Project.model;
+
+
+import org.hibernate.annotations.LazyCollection;
+import org.hibernate.annotations.LazyCollectionOption;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity
+@Table (name = "table")
+public class Match {
+
+    //------------------------
+    // ATTRIBUTES
+    //------------------------
+
+    @Id
+    private String matchId;
+
+    private String winnerId;
+    private String winnerName;
+    private GameMode gameMode;
+    private Options options;
+
+
+    //------------------------
+    // ASSOCIATIONS
+    //------------------------
+
+    @ManyToOne(optional = false)
+    private Account account;
+
+    @ManyToOne(optional = false)
+    private Player player;
+
+    @LazyCollection(LazyCollectionOption.FALSE)
+    @OneToMany(mappedBy = "match")
+    private List<Round> rounds;
+
+
+    //------------------------
+    // CONSTRUCTOR
+    //------------------------
+
+    public Match() {
+        matchId = null;
+        winnerId = null;
+        winnerName = null;
+        gameMode = null;
+        options = null;
+        account = null;
+        player = null;
+        rounds = new ArrayList<>();
+    }
+
+    public Match(String aMatchId, String aWinnerId, String aWinnerName, GameMode aGameMode, Options someOptions, Account anAccount, Player aPlayer) {
```

Comment:
also don't we need a list of players?

Reply:
Yeah, that would make sense but then we need to change the UML model